app/models/manageiq/providers/autosde/inventory/parser/storage_manager.rb

```diff
@@ -43,6 +45,54 @@ def storage_resources
     end
   end
 
+  def san_addresses
+    port_type = {
+      "ISCSI"  => "IscsiAddress",
+      "FC"     => "FiberChannelAddress",
+      "NVMeFC" => "NvmeAddress"
+    }
+
+    san_addresses = collector.storage_hosts.flat_map do |host_initiator|
```

Comment:
So now you don't even need to map the intermediate parsed hash, you can build it in one shot:
```ruby
# Collector
def storage_host_addresses
  @storage_host_addresses ||= storage_hosts.flat_map(&:addresses)
end

# Parser
collector.storage_host_addresses.each do |address|
  persister.san_addresses.build(
    :ems_ref     => address.uuid,
    :owner       => persister.host_initiators.lazy_find(host_initiator.uuid),
    :type        => port_type[address.port_type],
    :iqn         => address.iqn,
    :chap_name   => address.chap_name,
    :chap_secret => address.chap_secret,
    :wwpn        => address.wwpn
  )
end
```

This saves you from having the whole delete the `:host_initiators_uuid` and add back the whole `**san_address_hash` plus the `:owner` lazy fine

Reply:
@agrare  great suggestion, but I still need the info about the host_initiator (not just the address info) for the owner value, so I did it a bit different then how u suggested.

Reply:
https://github.com/ManageIQ/manageiq-providers-autosde/pull/49/files#diff-e094665ad11140a16079e84958bf4cbfddefeffd1ee276bc59a7a62e67ddd089R55-R68

Reply:
:+1: nice